### PR TITLE
S05US08

### PR DESCRIPTION
**Pull request by Chikoulat**
FEAT: Added argon to hash password, JWT and z for validation. 
FEAT: Added authentification and a login page.
FIX: Fixed an error in a navbar where all the link had a onClick to disconnect.

### CODE REVIEW

File: frontend/src/components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          navigate(`/candidat/profil/${auth.id}`);
+        });
+    } catch (error) {
+      setErr(error.response?.data?.message);
+    }
+  };
+  return (
+    <div className={`${style.profilesection} ${style.banner}`}>
+      <div className={{}}>
+        <h3 className={`${style.h3}`}>Bienvenu sur Externatic</h3>
```
Comment (YOUNS28100) [inline frontend/src/components/Login.jsx]: Bienvenue

File: frontend/src/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        <p className={{}}>Connecte toi</p>
+
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div className={{}}>
+            <input
+              className={{}}
+              type="email"
+              placeholder="Adresse mail"
+              {...register("email", { required: "L'e-mail est obligatioire" })}
```
Comment (YOUNS28100) [inline frontend/src/components/Login.jsx]: obligatoire

File: frontend/src/components/Login.jsx
Comment (YOUNS28100) [inline frontend/src/components/Login.jsx:1]: Plusieurs className vide dans le return

File: frontend/src/components/SideSection.jsx
Comment (YOUNS28100) [inline frontend/src/components/SideSection.jsx:1]: Pas d'accents dans les paths des NavLink

File: frontend/src/components/SideSection.jsx [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
-      <NavLink to="/candidat/compétence/20">
+      <NavLink to={`/candidat/profil/${user.id}`}>Compte</NavLink>
+      <NavLink to={`/candidat/diplôme/${user.id}`}>Mes diplômes</NavLink>
+      <NavLink to={`/candidat/expérience/${user.id}`}>Mes expérience</NavLink>
+      <NavLink to={`/candidat/compétence/${user.id}`}>
         Mes critères et compétences
       </NavLink>
-      <NavLink to="/candidat/messages/20"> Messages</NavLink>
+      <NavLink to={`/candidat/messages/${user.id}`}> Messages</NavLink>
       <NavLink> Suivi de candidatures </NavLink>
```
Comment (YOUNS28100) [inline frontend/src/components/SideSection.jsx:18]: Pas de path ?

File: frontend/src/components/sidebar/SideNavBar.jsx [hunk 4]
```diff
@@ -38,5 +62,7 @@ function SideBar({ sidebar, showSidebar }) {
 SideBar.propTypes = {
   showSidebar: PropTypes.func.isRequired,
   sidebar: PropTypes.bool.isRequired,
+  setAuth: PropTypes.func.isRequired,
+  auth: PropTypes.string.isRequired,
```
Comment (YOUNS28100) [inline frontend/src/components/sidebar/SideNavBar.jsx]: PropTypes.shape.isRequired,

File: frontend/src/components/SideSection.jsx [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
+      <NavLink to={`/candidat/activités/${user.id}`}>
+        Historique de candidatures
+      </NavLink>
     </section>
   );
 }
 
+SideSection.propTypes = {
+  user: PropTypes.shape({
+    id: PropTypes.string.isRequired,
```
Comment (YOUNS28100) [inline frontend/src/components/SideSection.jsx]: Number ?

File: frontend/src/components/sidebar/sideBarData.jsx [hunk 6]
```diff
@@ -25,16 +24,10 @@ const sidebarData = [
   },
   {
     title: "Candidat",
-    path: "/candidat/profil/11",
+    path: "/candidat/profil/15",
```
Comment (YOUNS28100) [inline frontend/src/components/sidebar/sideBarData.jsx]: dynamic id ?

File: frontend/src/components/user/CriteriaUser.jsx [hunk 7]
```diff
@@ -1,9 +1,18 @@
...
+  const [selectedValue, setSelectedValue] = useState("");
+
+  const handleSelectChange = (event) => {
+    setSelectedValue(event.target.value);
+  };
+
   return (
-    <select>
+    <select value={selectedValue} onChange={handleSelectChange}>
+      <option value="" disabled>
```
Comment (YOUNS28100) [inline frontend/src/components/user/CriteriaUser.jsx]: disabled ?

File: frontend/src/components/user/SkillsUser.jsx [hunk 8]
```diff
@@ -10,13 +10,12 @@ function SkillsUser() {
...
   return (
     <div>
-      {skills &&
-        skills.map((s) => (
-          <ul key={s.id}>
-            <li> {s.name} </li>
-            <li> {s.level} </li>
-          </ul>
-        ))}
+      {skills.map((s) => (
```
Comment (YOUNS28100) [inline frontend/src/components/user/SkillsUser.jsx]: skills &&

**Review by YOUNS28100 — CHANGES_REQUESTED** [review]
some comments

**Review by sdiakhate — APPROVED** [review]

**Review by ioayoub — APPROVED** [review]

**Pull request merged**